heudiconv/main.py

```diff
@@ -0,0 +1,363 @@
+    command : {'heuristics', 'heuristic-info', 'ls', 'populate-templates',
+               'sanitize-jsons', 'treat-jsons', None}, optional
+        Custom action to be performed on provided files instead of regular
+        operation. Default is None.
+    grouping : {'studyUID', 'accession_number', 'all', 'custom'}, optional
+        How to group dicoms. Default is 'studyUID'.
+    minmeta : bool, optional
+        Exclude dcmstack meta information in sidecar jsons. Default is False.
+    random_seed : int or None, optional
+        Random seed to initialize RNG. Default is None.
+    dcmconfig : str or None, optional
+        JSON file for additional dcm2niix configuration. Default is None.
+    queue : {'SLURM', None}, optional
+        Batch system to submit jobs in parallel. Default is None.
+    queue_args : str or None, optional
+        Additional queue arguments passed as single string of space-separated
+        Argument=Value pairs. Default is None.
+
+    Notes
+    -----
+    All parameters in this function must be called as keyword arguments.
+    """
+
+    # To be done asap so anything random is deterministic
+    if random_seed is not None:
+        import random
+        random.seed(random_seed)
+        import numpy
+        numpy.random.seed(random_seed)
+    # Ensure only supported bids options are passed
+    if debug:
+        lgr.setLevel(logging.DEBUG)
+    # Should be possible but only with a single subject -- will be used to
+    # override subject deduced from the DICOMs
+    if files and subjs and len(subjs) > 1:
+        raise ValueError(
+            "Unable to processes multiple `--subjects` with files"
+        )
+
+    if debug:
+        setup_exceptionhook()
+
+    # Deal with provided files or templates
+    # pre-process provided list of files and possibly sort into groups/sessions
+    # Group files per each study/sid/session
+
+    outdir = op.abspath(outdir)
+
+    try:
+        import etelemetry
+        latest = etelemetry.get_project("nipy/heudiconv")
+    except Exception as e:
+        lgr.warning("Could not check for version updates: %s", str(e))
+        latest = {"version": 'Unknown'}
+
+    lgr.info(INIT_MSG(packname=__packagename__,
+                      version=__version__,
+                      latest=latest["version"]))
+
+    if command:
+        process_extra_commands(outdir, command, files, dicom_dir_template,
+                               heuristic, session, subjs, grouping)
+        return
+    #
+    # Load heuristic -- better do it asap to make sure it loads correctly
+    #
+    if not heuristic:
+        raise RuntimeError("No heuristic specified - add to arguments and rerun")
+
+    if queue:
+        lgr.info("Queuing %s conversion", queue)
+        iterarg, iterables = ("files", len(files)) if files else \
+                             ("subjects", len(subjs))
+        queue_conversion(queue, iterarg, iterables, queue_args)
+        sys.exit(0)
```

Comment:
I think we must not `sys.exit(0)` here now -- I will RF so we exit in the CLI invocation